src/nvim/CMakeLists.txt

```diff
@@ -176,9 +170,16 @@ list(APPEND NVIM_LINK_LIBRARIES
     ${CMAKE_THREAD_LIBS_INIT}
     )
 
+set(NVIM_EXEC_LINK_LIBRARIES ${NVIM_LINK_LIBRARIES})
+
+if(USE_JEMALLOC)
+  # dont use jemalloc in the unit test library
+  list(APPEND NVIM_EXEC_LINK_LIBRARIES ${JEMALLOC_LIBRARIES})
```

Comment:
Unit tests often trade memory chunks between luajit and C, and making luajit aware of jemalloc would be more troublesome.

I've read reports such as [this one](https://bugzilla.mozilla.org/show_bug.cgi?id=880193) suggesting that ASAN chokes with jemalloc, so I'd rather not introduce that risk for us. 

Besides, since the main goal of using jemalloc is gaining performance, it's kinda pointless to use it with unit tests or ASAN.